src/Credentials.js

```diff
@@ -253,25 +356,68 @@ class Credentials {
-    return createJWT(this.settings, {sub: sub, claim, exp})
+    return this.signJWT({sub: sub, claim, exp})
   }
 
-/**
-  *  Look up a profile in the registry for a given uPort address. Address must be MNID encoded.
-  *
-  *  @example
-  *  credentials.lookup('5A8bRWU3F7j3REx3vkJ...').then(profile => {
-  *     const name = profile.name
-  *     const pubkey = profile.pubkey
-  *     ...
-  *   })
+  /**
+  *  Builds and returns a contract object which can be used to interact with
+  *  a given contract. Similar to web3.eth.contract but with promises. Once specifying .at(address)
+  *  you can call the contract functions with this object. Each call will create a request.
   *
-  * @param    {String}            address             a MNID encoded address
-  * @return   {Promise<Object, Error>}                a promise which resolves with parsed profile or rejects with an error
+  *  @param    {Object}       abi                                   contract ABI
+  *  @return   {Object}                                             contract object
   */
-  lookup (address) {
-    return this.settings.registry(address)
+  contract (abi) {
+    const txObjHandler = (txObj, opts) => {
+      txObj.fn = txObj.function
+      delete txObj['function']
+      return this.txRequest(txObj, opts)
+    }
+    return ContractFactory(txObjHandler.bind(this))(abi)
+  }
+
+  /**
+   *  Given a transaction object, similarly defined as the web3 transaction object,
+   *  it creates a JWT transaction request and appends addtional request options.
+   *
+   *  @example
+   *  const txobject = {
+   *    to: '0xc3245e75d3ecd1e81a9bfb6558b6dafe71e9f347',
+   *    value: '0.1',
+   *    fn: "setStatus(string 'hello', bytes32 '0xc3245e75d3ecd1e81a9bfb6558b6dafe71e9f347')",
+   *  }
+   *  connect.txRequest(txObject, {callbackUrl: 'http://mycb.domain'}).then(jwt => {
+   *    ...
+   *  })
+   *
+   *  @param    {Object}    txObj
+   *  @param    {String}    [id='addressReq']    string to identify request, later used to get response
+   *  @return   {String}                         a transaction request jwt
+   */
+  txRequest(txObj, { callbackUrl, exp = 600, network_id, label } = {}) {
+    const payload = {}
+    if (callbackUrl) payload.callback = callbackUrl
+    if (network_id) payload.net = network_id
+    if (label) payload.label = label
+    return this.signJWT({...payload, ...txObj, type: 'ethtx'}, exp )
   }
+
+// /**
+//   *  Look up a profile in the registry for a given uPort address. Address must be MNID encoded.
+//   *
+//   *  @example
+//   *  credentials.lookup('5A8bRWU3F7j3REx3vkJ...').then(profile => {
+//   *     const name = profile.name
+//   *     const pubkey = profile.pubkey
+//   *     ...
+//   *   })
+//   *
+//   * @param    {String}            address             a MNID encoded address
+//   * @return   {Promise<Object, Error>}                a promise which resolves with parsed profile or rejects with an error
+//   */
+//   lookup (address) {
```

Comment:
thinking here? keeping this?

Reply:
No. It's handled by a DID resolver now. Lets remove the commented out section